### DISCUSSION TITLE
Making the app work without granting shared libraries permissions

### DISCUSSION THREAD

**Discussion by antoniodiazroman** [abraunegg/onedrive#2613] [Q&A] [answered]
Hello,

First and foremost, thank you for your excellent work!

I'm using Ubuntu at work and I'm considering to use this application to synchronize my work data with my OneDrive personal space that my company provides to me. I asked the IT Department to analyze the application to find out wheter or not it could be used with this aim in mind.

While being tested, the IT Department realized that the application asks for permissions to access all the Sharepoint libraries, I guess with the objective to provide their content through the shared libraries menu of the application. However, this access request has not been approved arguing that it's a security issue, something that I utterly understand.

So, in the light of the abovementioned consideration, I would like to know if there is a way to run the application without granting it the permissions to the shared libraries it requests, maybe disabling the shared libraries menu, or something like that.

Thank you very much in advance!

**Comment by abraunegg** [comment, marked as answer]
@antoniodiazroman 
There is no way around the granting of permissions - this is how API and authenticated API's work.

Please have your IT Team read this: https://github.com/abraunegg/onedrive/blob/master/docs/application-security.md
